sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -274,12 +275,30 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
   }
 
   object CartesianProduct extends Strategy {
+    def getSmallSide(left: LogicalPlan, right: LogicalPlan): BuildSide = {
+      if (right.statistics.sizeInBytes < left.statistics.sizeInBytes) {
+        joins.BuildRight
+      } else {
+        joins.BuildLeft
+      }
+    }
```

Comment:
I am not sure comparing the total size will give us a very useful heuristics because the size of a table does not imply anything about the size of a partition. Also, I am inclined to get https://github.com/apache/spark/pull/8652/files merged first and then add any other optimization in a follow-up PR. What do you think?